pkg/apis/options/load.go

```diff
@@ -55,6 +55,87 @@ func Load(configFileName string, flagSet *pflag.FlagSet, into interface{}) error
 	return nil
 }
 
+// LoadYAML will load a YAML based configuration file into the options interface provided.
+func LoadYAML(configFileName string, opts interface{}) error {
+	buffer, err := loadAndSubstituteEnvs(configFileName)
+	if err != nil {
+		return err
+	}
+
+	// Generic interface for loading arbitrary yaml structure
+	var intermediate map[string]interface{}
+
+	if err := yaml.Unmarshal(buffer, &intermediate); err != nil {
+		return fmt.Errorf("error unmarshalling config: %w", err)
+	}
+
+	// Using mapstructure to decode arbitrary yaml structure into options and
+	// merge with existing values instead of overwriting everything. This is especially
+	// important as we have a lot of default values for boolean which are supposed to be
+	// true by default. Normally by just parsing through yaml all booleans that aren't
+	// referenced in the config file would be parsed as false and we cannot identify after
+	// the fact if they have been explicitly set to false or have not been referenced.
+	return Decode(intermediate, opts)
+}
+
+// Decode processes an input map and decodes it into a given struct while preserving default values.
+// It ensures proper conversion of duration values from strings, floats, and int64 into time.Duration.
+//
+// Parameters:
+// - input: A map[string]interface{} representing the input data.
+// - result: A pointer to a struct where the decoded values will be stored.
+//
+// Returns:
+// - An error if decoding fails or if there are unmapped keys.
+func Decode(input interface{}, result interface{}) error {
+	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		DecodeHook: mapstructure.ComposeDecodeHookFunc(
+			toDurationHookFunc(),
+			stringToBytesHookFunc(),
+		),
+		Metadata:             nil,    // Don't track any metadata
+		Result:               result, // Decode the result into the prefilled options
+		TagName:              "yaml", // Parse all fields that use the json tag
```

Comment:
Is there a reason to switch to yaml instead of json? Is some other library relying on yaml?

Reply:
Ack, can you just explain in the comment here why we use the yaml tag

Some yaml libraries like the `json` tag instead